core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -303,13 +308,23 @@ public void onHeaders(SdkHttpResponse headers) {
             long d = now - startTime;
             context.attemptMetricCollector().reportMetric(CoreMetric.TIME_TO_FIRST_BYTE, Duration.ofNanos(d));
             super.onHeaders(headers);
+
+            context.progressUpdater().ifPresent(progressUpdater -> {
+                ProgressListenerUtils.updateProgressListenersWithResponseStatus(progressUpdater, headers);
+            });
         }
 
         @Override
         public void onStream(Publisher<ByteBuffer> stream) {
             AtomicLong bytesReadCounter = context.executionAttributes()
                                                  .getAttribute(SdkInternalExecutionAttribute.RESPONSE_BYTES_READ);
-            BytesReadTrackingPublisher bytesReadTrackingPublisher = new BytesReadTrackingPublisher(stream, bytesReadCounter);
+
+            ProgressUpdater progressUpdater = context.progressUpdater().isPresent() ?
```

Comment:
nit: can be shortened with `context.progressUpdater().orElse(null)`

Reply:
done, I used a NoOp listener to do away with this check completely.